aten/src/ATen/detail/HIPHooksInterface.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/detail/HIPHooksInterface.h>
+
+#include <c10/util/Exception.h>
+
+#include <cstddef>
+#include <memory>
+#include <mutex>
+
+namespace at {
+namespace detail {
+
+const HIPHooksInterface& getHIPHooks() {
+  static std::unique_ptr<HIPHooksInterface> hip_hooks;
+  // NB: The once_flag here implies that if you try to call any HIP
+  // functionality before libATen_cuda.so is loaded, HIP is permanently
```

Comment:
perhaps you should clarify the libATen_cuda.so situation in the comment (are we going to have a libHIP?)

Reply:
I just deleted this comment XD

Reply:
But yes, we'll have a libATen_hip once we get out of place hipify